rlberry/utils/check_agent.py

```diff
@@ -0,0 +1,202 @@
+from rlberry.envs import Chain
+from rlberry.envs.benchmarks.ball_exploration import PBall2D
+from rlberry.manager import AgentManager
+import numpy as np
+from rlberry.seeding import set_external_seed
+import tempfile
+import os
+
+SEED = 42
+
+
+def check_agent_manager(Agent, continuous_state=False):
```

Comment:
I see that the flag `continuous_state` is only used to get a constructor for the environment. I think it's better to replace it by a tuple `(env_constructor, env_kwargs)`, so that the users can test their own combinations of agent+environment.

I'm saying this because there are many observation spaces that aren't continuous or discrete. We can have `Dict` spaces with possibly nested continuous/discrete observations, and currently the `check_agent_manager` method (and others) assume either a `Discrete` or `Box` observation space.

Reply:
Ok but at the same time my check aimed at being simple and if I add what you suggest then the user can't just `check_rl_agent(My_awesome_agent)`, he has to also define an environment. And also I prefered to do this because I am sure that Chain and PBall2D are fine, if the user has a problem he will not know whether this is due to the agent or the environment. 
But at the same time you are right that more generality would be nice. Maybe I can have the two possible parameters : if env is given, use it if not use default env.

Reply:
Alright! But then there is some redundancy if we keep both `env` and `continuous_state` as parameters. What do you think of using a single parameter `env` that can be either:
* a string `continuous_state`
* a string `discrete_state`
* a tuple (constructor, kwargs)
?

It would also allow us to add extra options later, as `continuous_state_action` etc.